lib/SPIRV/SPIRVReader.cpp

```diff
@@ -354,6 +354,17 @@ std::string SPIRVToLLVM::transVCTypeName(SPIRVTypeBufferSurfaceINTEL *PST) {
     return VectorComputeUtil::getVCBufferSurfaceName(PST->getAccessQualifier());
   return VectorComputeUtil::getVCBufferSurfaceName();
 }
+static unsigned transPointerStorageClass(SPIRVTypePointer const *PtrType) {
+  assert(PtrType && "Must be valid pointer type.");
+
+  if (PtrType->getPointerElementType()->getOpCode() == OpTypeFunction) {
+    auto *M = PtrType->getModule();
+    if (M->hasProgramAddressSpace())
```

Comment:
It seems to me, that any function in the module will have the set address space, which seems incorrect to me (just consider such artificial example:
```
%ptr = alloca i8
call addrspace(3) void @llvm.lifetime.start.p0i8(i64 1, i8* %ptr)
```
how backend will consider addrspace(3) ?
What if a function had address space before the translation? Not this AS is just being lost (and some BE might do some assumptions regarding what AS to pick), with this option turned on the AS won't do any assumptions, just picking the set AS, is it error-prune?

Reply:
If the function had addrspace before the translation the current translator will preserve AS on a pointer but will lose it on the function itself, which will result in an error during the compilation. To some extend this options saves us. Though fixing the current function pointer implementation first is a nice idea too. There's CodeSectionINTEL storage class but it is never used and it is not clear what LLVM IR addrspace should it correspond.